js/compiler/compiler.go

```diff
@@ -79,28 +79,45 @@ var (
 		"highlightCode": false,
 	}
 
-	once        sync.Once // nolint:gochecknoglobals
-	globalBabel *babel    // nolint:gochecknoglobals
+	onceBabelCode   sync.Once     // nolint:gochecknoglobals
+	globalBabelCode *goja.Program // nolint:gochecknoglobals
+	onceBabel       sync.Once     // nolint:gochecknoglobals
+	globalBabel     *babel        // nolint:gochecknoglobals
 )
 
 // A Compiler compiles JavaScript source code (ES5.1 or ES6) into a goja.Program
 type Compiler struct {
 	logger logrus.FieldLogger
+	babel  *babel
 }
 
 // New returns a new Compiler
 func New(logger logrus.FieldLogger) *Compiler {
 	return &Compiler{logger: logger}
 }
 
+// InitializeBabel specifically initializes a separate (non-global) instance of babel for this compiler specifically
+// the returned error is only if babel couldn't be parsed/run which should never be possible
+func (c *Compiler) InitializeBabel() error {
+	var err error
+	if c.babel == nil {
+		c.babel, err = newBabel()
+	}
+	return err
+}
+
 // Transform the given code into ES5
 func (c *Compiler) Transform(src, filename string) (code string, srcmap *SourceMap, err error) {
-	var b *babel
-	if b, err = newBabel(); err != nil {
-		return
+	if c.babel == nil {
+		onceBabel.Do(func() {
+			globalBabel, err = newBabel()
+		})
+		c.babel = globalBabel
```

Comment:
It should never happen, but should we still check and return `err` here?

Reply:
hm ... I thought I did at least on the first ...uuh .. return.

Reply:
fixed it